libcloud/compute/drivers/gce.py

```diff
@@ -3843,15 +3863,23 @@ def ex_create_network(self, name, cidr, description=None, mode="legacy"):
         if cidr and mode in ['auto', 'custom']:
             raise ValueError("Can only specify IPv4Range with 'legacy' mode.")
 
-        request = '/global/networks'
-
         if mode == 'legacy':
             if not cidr:
                 raise ValueError("Must specify IPv4Range with 'legacy' mode.")
             network_data['IPv4Range'] = cidr
         else:
             network_data['autoCreateSubnetworks'] = (mode.lower() == 'auto')
 
+        if routing_mode.lower() not in ['regional', 'global']:
```

Comment:
I think this will fail when `routing_mode` is `None` (default value)?

Reply:
I didn't get failures in testing w/o a value (would end up passing blank routingConfig{} data), but how about I just rejigger this a bit by only triggering logic if routing_mode is specified?
```python
        if routing_mode:
            if routing_mode.lower() not in ['regional', 'global']:
                raise ValueError("Invalid Routing Mode: '%s'. Must be "
                                 "'REGIONAL' or 'GLOBAL'." % routing_mode)
            else:
                network_data['routingConfig'] = {
                    'routingMode': routing_mode.upper()
                }
```

Reply:
The change looks good, but you're saying it's unnecessary? I don't understand, if routing_mode is None, `routing_mode.lower()` is going to raise an exception, right? Am I missing missing?